pyaedt/edb_core/edb_data/siwave_simulation_setup_data.py

```diff
@@ -0,0 +1,897 @@
+        self._edb_sim_setup_info.SimulationSettings.UseSISettings = value
+        self._update_setup()
+
+    @pyaedt_function_handler()
+    def add_frequency_sweep(self, name=None, frequency_sweep=None):
+        """Add frequency sweep.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the frequency sweep.
+        frequency_sweep : list, optional
+
+        Returns
+        ----------
+        :class:`pyaedt.edb_core.edb_data.simulation_setup_data.EdbFrequencySweep`
+        """
+        if name in self.frequency_sweeps:
+            return False
+        if not name:
+            name = generate_unique_name("sweep")
+        sweep = EdbFrequencySweep(self, frequency_sweep, name)
+        self._sweep_data_list[name] = sweep
+        return sweep
+
+
+class SiwaveDCSimulationSetup(SiwaveDCAdvancedSettings, object):
+    """Manages EDB methods for hfss simulation setup."""
+
+    def __init__(self, edb, name=None, edb_siwave_sim_setup=None):
+        self._edb = edb
+        self._mesh_operations = {}
+
+        if edb_siwave_sim_setup:
+            self._edb_sim_setup_info = edb_siwave_sim_setup
+        else:
+            self._edb_sim_setup_info = self._edb.simsetupdata.SimSetupInfo[
+                self._edb.simsetupdata.SIwave.SIWDCIRSimulationSettings
+            ]()
+            if not name:
+                self._edb_sim_setup_info.Name = generate_unique_name("siwave")
+            else:
+                self._edb_sim_setup_info.Name = name
+            self._update_setup()
+        SiwaveDCAdvancedSettings.__init__(self, self)
+
+    @property
+    def edb_sim_setup_info(self):
+        """Edb Internal Simulation Setup Object."""
+        return self._edb_sim_setup_info
+
+    @pyaedt_function_handler()
+    def _update_setup(self):
+        edb_sim_setup = self._edb.edb.Utility.SIWaveDCIRSimulationSetup(self._edb_sim_setup_info)
+        if self.name in self._edb.setups:
+            self._edb._active_layout.GetCell().DeleteSimulationSetup(self.name)
+        self._edb.active_cell.AddSimulationSetup(edb_sim_setup)
+        return True
+
+    @property
+    def name(self):
+        """Get name of this setup."""
+        return self._edb_sim_setup_info.Name
+
+    @name.setter
+    def name(self, value):
+        """Set name of this setup."""
+        self._edb_sim_setup_info.Name = value
+        self._update_setup()
+
+    @property
+    def enabled(self):
+        """Check if setup is enabled."""
+        return self._edb_sim_setup_info.SimulationSettings.Enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        self._edb_sim_setup_info.SimulationSettings.Enabled = value
+        self._update_setup()
+
+    @property
+    def source_terms_to_ground(self):
+        """Retrieve the dictionary of grounded terminals.
+
+        Returns
+        -------
+            Dictionary
+            {str, int}, keys is source name, value int 0 unspecified, 1 negative node, 2 positive one.
```

Comment:
This doesn't seem to be a standard way of describing. It should be described in an initial fragment followed by additional information in a compete sentence.